### PR TITLE
[Build] Allow building without Conan

### LINKED ISSUE #75: Make Conan optional
**Describe the bug**
Conan is not available on most Linux distributions and is not really not necessary to build this since Linux has a working C library management system by itself.

**To Reproduce**
1. create a new empty build directory in the git checkout
2. run `cmake ..` in this directory
3. See error
```
CMake Error at cmake-proxies/cmake-modules/conan.cmake:795 (message):
  Conan executable not found! Please install conan.
Call Stack (most recent call first):
  cmake-proxies/cmake-modules/conan.cmake:841 (conan_check)
  cmake-proxies/cmake-modules/AudacityDependencies.cmake:4 (conan_add_remote)
  CMakeLists.txt:154 (include)
```
**Expected behavior**
It build with the libraries provided by the system without Conan.

**Additional information (please complete the following information):**
 - OS: Arch Linux

### PR DESCRIPTION
**Pull request by AnotherFoxGuy**
fixes: #75

PR for building without having Conan installed

STATUS:
Builds and runs on Manjaro with `wxgtk3-dev 3.1.5-2` (See: https://github.com/tenacityteam/tenacity/pull/234#issuecomment-884876066)

<details>
<summary>Contributor stuff</summary>

:white_check_mark: The code in this pull request is licensed under "GPLv2 or any later version"\*
:white_check_mark: I made sure the code compiles on my machine
:white_check_mark: I made sure there are no unnecessary changes in the code\*
:white_check_mark: I made sure the title of the PR reflects the core meaning of the issue you are solving\*
:white_check_mark: I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"\*

</details>

### CODE REVIEW

**Comment by mariaWitch** [comment]
This is placebo levels of removal. Disabling conan != removing it.

**Comment by Be-ing** [comment]
Thanks for working on this but I am in doubt this is a good direction to continue. Audacity is planning [more changes](https://github.com/audacity/audacity/issues/1187) to their dependency management, so I think it may be difficult to maintain this in the future. Judging from what they have done so far and their continued snide remarks towards Linux users I have little confidence those changes will turn out very well. I think it will be easier to maintain to bypass the Conan mess entirely as #228 does.

**Comment by nbsp** [comment]
What is the status of this PR? #228 has made significant progress on the same front. If this is no longer actively developed, please close this PR.

**Comment by n0toose** [comment]
To be honest, it could be good if the `USE_CONAN` variable was still there, so some of his (much appreciated, BTW) efforts can still be used, carried on and adjusted to the more recent/newer changes.

**Comment by AnotherFoxGuy** [comment]
> What is the status of this PR?

It builds with wxWidgets V3.1.5, but I haven't tested if it runs on linux

**Comment by n0toose** [comment]
This may be very important for our first release. See: https://github.com/tenacityteam/tenacity/pull/300#issuecomment-884782522

**Comment by AnotherFoxGuy** [comment]
Just tested this on manjaro with `wxgtk3-dev 3.1.5-2`, seems to work fine
Screenshot_20210722_142952

**Review by n0toose — DISMISSED** [review]

**Comment by nbsp** [comment]
Could this be rebased into one commit `Allow building without Conan`? I doubt the bug fix warrants its own commit

**Comment by n0toose** [comment]
FYI: 🎉  emojis look good on the GitHub front-end but weird on the Git CLI and other frontends.

I'd suggest changing the `:tada:` to an actual emoji or getting rid of it altogether?

**Comment by n0toose** [comment]
Building this without Conan seems to work.

**Review by n0toose — DISMISSED** [review]
Seems good to me, I think building this with Conan using the correct parameter needs some further testing.

File: cmake-proxies/CMakeLists.txt [hunk 1]
```diff
@@ -81,15 +81,15 @@ add_conan_lib(
...
    INTERFACE_NAME libmp3lame::libmp3lame
-   PKG_CONFIG "lame >= 3.100"
 )
 
 add_conan_lib(
    libid3tag
    libid3tag/0.15.2b@audacity/stable
    OPTION_NAME libid3tag
-   PKG_CONFIG "id3tag >= 0.15.1b"
+   PKG_CONFIG "id3tag >= 0.15.1b"
```
Comment (n0toose) [inline cmake-proxies/CMakeLists.txt]: Trailing space here?

**Review by Be-ing — DISMISSED** [review]
Please rename Findlibmp3lame.cmake to Findmp3lame.cmake for consistency with the CMake target name in vcpkg.

**Comment by AnotherFoxGuy** [comment]
> @Be-ing Please rename Findlibmp3lame.cmake to Findmp3lame.cmake for consistency with the CMake target name in vcpkg.

Renaming lame will completely break the buildsystem, I will open a separate PR to fix it 
(Also, vcpkg is the only one that calls it `mp3lame`)

**Review by fossdd — APPROVED** [review]
Builds and works.

**Review by n0toose — APPROVED** [review]
Edgar, this seriously rocks.

**Pull request merged**